### PR TITLE
List: Adding some list improvements

### PR DESCRIPTION
**Pull request by otaviobp**

### CODE REVIEW

File: src/lib/list.h [hunk 1]
```diff
@@ -125,41 +155,19 @@ bool list_longer_than(struct list *list, int count);
...
- * - list1 and list2 are sorted
- * - the comparison_fn should behave similar to strcmp(), returning 0 if it's a match, and "< 0" or "> 0" if it's not
+ * @brief Filters any element from sorted list list1 that happens to be in the sorted list list2 meeting the criteria.
  */
-struct list *list_filter_common_elements(struct list *list1, struct list *list2, comparison_fn_t comparison_fn, list_free_data_fn_t list_free_data_fn);
+struct list *list_sorted_filter_common_elements(struct list *list1, struct list *list2, comparison_fn_t comparison_fn, list_free_data_fn_t list_free_data_fn);
 
 /**
- * @brief sorts and removes all duplicates from a list of strings
+ * @brief removes the first occurrence of an item and returs a pointer to its data.
```
Comment (karthikprabhuvinod) [inline src/lib/list.h]: typo: returns

File: src/lib/list.h [hunk 2]
```diff
@@ -36,6 +36,9 @@ typedef void (*list_free_data_fn_t)(void *data);
 /** @brief Callback to filter a data in a list. */
 typedef bool (*filter_fn_t)(const void *a);
 
+/** @brief Callback to cone a data in a list. */
```
Comment (karthikprabhuvinod) [inline src/lib/list.h]: typo: clone

File: src/lib/list.c [hunk 3]
```diff
@@ -205,20 +205,21 @@ struct list *list_concat(struct list *list1, struct list *list2)
 	struct list *tail;
 
 	list2 = list_head(list2);
+	tail = list_tail(list1);
```
Comment (karthikprabhuvinod) [inline src/lib/list.c]: maybe name this tail1

File: src/lib/list.h [hunk 4]
```diff
@@ -4,6 +4,17 @@
 /**
  * @file
  * @brief Doubly linked list implementation.
+ *
+ * All list functions that performs operation in all nodes should always require
+ * that any element of the list is used as a parameter and return the first
+ * element of the list. Caller is not required to use list_head() in the
+ * parameter or result.
```
Comment (castulo) [inline src/lib/list.h]: 👍

File: src/lib/list.h [hunk 5]
```diff
@@ -40,15 +51,26 @@ typedef bool (*filter_fn_t)(const void *a);
...
+ * new elements without having to traverse the whole list to find the last one.
  */
 struct list *list_append_data(struct list *list, void *data);
 
 /**
  * @brief Prepend one element to the list.
+ * @param list any element of the list
+ * @param data the data the new element should keep
+ *
+ * @returns the prepended elemement or NULL on failures.
```
Comment (karthikprabhuvinod) [inline src/lib/list.h]: isnt the prepended element the head of the list? Also element spelt wrong

Comment (otaviobp) [reply src/lib/list.h]: That's true. The prepended is always the first element of the list.

**Pull request merged**